### PR TITLE
Add raycast shooting with basic effects

### PR DESCRIPTION
**Pull request by MadStark**
image

### CODE REVIEW

File: Assets/Scripts/Extensions/GameObjectExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+﻿using UnityEngine;
+
+namespace Bug
```
Comment (Xwilarg) [inline Assets/Scripts/Extensions/GameObjectExtensions.cs:3]: I think namespace should be Bug.Extensions

File: Assets/Scripts/Player/PlayerController.cs [hunk 2]
```diff
@@ -218,36 +223,50 @@ public void OnLook(InputAction.CallbackContext value)
...
+			// 		_armsAnimator.SetTrigger(_animShootTriggerHash);
+			// 	}
+			//
+			// 	_slotWeapons[(int)_currentWeapon].AmmoInGun--;
+			// 	if (_slotWeapons[(int)_currentWeapon].AmmoInGun == 0 && _slotWeapons[(int)_currentWeapon].NbOfMagazines > 0)
+			// 	{
+			// 		StartCoroutine(Reload());
+			// 	}
+			// 	UpdateAmmoDisplay();
+			// }
```
Comment (Xwilarg) [inline Assets/Scripts/Player/PlayerController.cs]: Not sure why you kept that if it's not useful anymore?

File: Assets/Scripts/Player/PlayerController.cs [hunk 3]
```diff
@@ -218,36 +223,50 @@ public void OnLook(InputAction.CallbackContext value)
...
 			}
 		}
 
 		public void OnShoot(InputAction.CallbackContext value)
 		{
-			if (value.phase == InputActionPhase.Performed && // Already done the action when the button is pressed
-				!IsGamePaused() && // Don't shoot if the game is paused
-				!_isReloading && // We can't do it if we are reloading
-				_slotWeapons[(int)_currentWeapon].AmmoInGun > 0 && // We can't do it if there are no ammo left
-				_carriedObject == null) // Can't do anything while we are carrying an object
```
Comment (Xwilarg) [inline Assets/Scripts/Player/PlayerController.cs:233]: Where did you move these checks?

**Review by Xwilarg — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Extensions/GameObjectExtensions.cs (hunk 1 above)
Comment (MadStark) [reply Assets/Scripts/Extensions/GameObjectExtensions.cs:3]: I thought of that, but I tend to not like it, extension methods should be easy to access, and remembering to add the using Extensions namespace slows down the process in my opinion!
I like to use namespaces to separate the code, so if I don't want to separate it, I'll go less deep in the namespace!

File: Assets/Scripts/Player/PlayerController.cs (hunk 3 above)
Comment (MadStark) [reply Assets/Scripts/Player/PlayerController.cs:233]: I moved the responsibility of the ammo counter and reloading time to the gun (or "held item"), and it is check by the PlayerController via `CanHandlePrimaryAction`.

File: Assets/Scripts/Extensions/GameObjectExtensions.cs (hunk 1 above)
Comment (Xwilarg) [reply Assets/Scripts/Extensions/GameObjectExtensions.cs:3]: Sounds good to me

File: Assets/Scripts/Player/PlayerController.cs (hunk 3 above)
Comment (Xwilarg) [reply Assets/Scripts/Player/PlayerController.cs:233]: carried item isn't the gun
It's when you are carrying an object (like a sentry)

**Review by Xwilarg — APPROVED** [review]

**Pull request merged**